src/sidebar/store/util.js

```diff
@@ -44,16 +44,19 @@ export function createReducer(actionToUpdateFn) {
  * selectors with the `state` argument set to the current value of `getState()`.
  */
 export function bindSelectors(namespaces, getState) {
-  const totalSelectors = {};
+  const boundSelectors = {};
   Object.keys(namespaces).forEach(namespace => {
-    const selectors = namespaces[namespace].selectors;
+    const { selectors, rootSelectors = {} } = namespaces[namespace];
+
     Object.keys(selectors).forEach(selector => {
-      totalSelectors[selector] = function () {
-        const args = [].slice.apply(arguments);
-        args.unshift(getState());
-        return selectors[selector].apply(null, args);
-      };
+      boundSelectors[selector] = (...args) =>
+        selectors[selector](getState()[namespace], ...args);
+    });
+
+    Object.keys(rootSelectors).forEach(selector => {
```

Comment:
What's the behavior on name collisions (duplicate in `rootSelector`)?

Reply:
If keys in `selectors` and `rootSelectors` overlap then the key in `rootSelectors` wins. Better behaviour might be to throw an error.

Reply:
Update - I've changed this code to throw an error if selectors conflict.